SemiconductorTestLibrary.TestStandSteps/source/LeakageTest.cs

```diff
@@ -67,64 +70,80 @@ public static void LeakageTest(
 
                 var originalSourceDelays = dcPower?.GetSourceDelayInSeconds();
                 var dcPowerMeasureSettings = new DCPowerMeasureSettings() { ApertureTime = apertureTime };
-                // Source specified voltage either sequentially or in parallel according to <paramref name="serialOperationEnabled"/>.
+
+                // Configure initial settings and Force 0 V on all pins.
+                InvokeInParallel(
+                    () =>
+                    {
+                        if (dcPower != null)
+                        {
+                            dcPower.ConfigureSourceDelay(settlingTime);
+                            dcPower.ConfigureMeasureSettings(dcPowerMeasureSettings);
+                            dcPower.ForceVoltage(0, currentLimit, waitForSourceCompletion: true);
+                        }
+                    },
+                    () =>
+                    {
+                        if (digital != null)
+                        {
+                            digital.ConfigureApertureTime(apertureTime);
+                            digital.ForceVoltage(0, currentLimit);
+                            PreciseWait(settlingTime);
+                        }
+                    });
+
+                // Force specified voltage either sequentially or in parallel according to <paramref name="serialOperationEnabled"/>.
                 if (serialOperationEnabled)
                 {
+                    // Force specified voltage, measure current, then disable the output on each pin or pin group in serially, one pin or pin group at a time.
                     foreach (var pinOrPinGroup in pinsOrPinGroups)
                     {
                         var dcPowerPins = tsmContext.FilterPinsByInstrumentType(new string[] { pinOrPinGroup }, InstrumentTypeIdConstants.NIDCPower);
                         if (dcPowerPins.Any())
                         {
                             var dcPowerPerPinOrPinGroup = sessionManager.DCPower(dcPowerPins);
-                            dcPowerPerPinOrPinGroup.ConfigureSourceDelay(settlingTime);
-                            dcPowerPerPinOrPinGroup.ConfigureMeasureSettings(dcPowerMeasureSettings);
-                            dcPowerPerPinOrPinGroup.ForceVoltage(voltageLevel, currentLimit);
+                            dcPowerPerPinOrPinGroup.ForceVoltage(voltageLevel, currentLimit, waitForSourceCompletion: true);
+                            dcPowerPerPinOrPinGroup.MeasureAndPublishVoltage("Leakage", out _);
+                            dcPowerPerPinOrPinGroup.PowerDown();
```

Comment:
Now that we are comparing to the legacy implementation, should we
1. Force to 0V before calling PowerDown?
2. Do PowerDown for all pins at once in the end?

Reply:
@lornazh, good point, I will push a commit now that we reviewed together that implements this.